File: archiV1java/inventory/src/main/java/om/inventory/svc/InventoryServiceImpl.java
package om.inventory.svc;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import om.inventory.dao.InventoryRepo;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor
@Service @Slf4j
public class InventoryServiceImpl implements InventoryService {
    private final InventoryRepo inventoryRepo;

    public boolean isItemInStock(String itemSkuCode, Integer quantityToCheck) {
        if (quantityToCheck > 0) {
            boolean available = inventoryRepo.existsByItemSkuCodeAndQuantityInStockIsGreaterThanEqual(itemSkuCode, quantityToCheck);
            if(available)
                log.info("SUCCESS: The requested quantity of the item (SKU: " + itemSkuCode + ") is available for preparing the requested order.");
            else
                log.info("FAILURE: The product item with SKU " + itemSkuCode + " is out of stock.");
            return available;
        }
        log.info("ERROR: Kindly place the next request with a decent quantity. Non-positive quantities are not acceptable at this moment.");
        return false;
    }
}

File: snapshotV2/inventory/src/main/kotlin/om/inventory/service/InventoryServiceImpl.kt
package om.inventory.service

import om.inventory.exception.StockLevelException
import om.inventory.repo.InventoryRepository
import org.springframework.stereotype.Service

@Service
class InventoryServiceImpl(val inventoryRepository: InventoryRepository):InventoryService {
    override fun isItemInStock(skuCode: String, quantityToCheck: Int):Boolean {
        if( quantityToCheck>0){
            return inventoryRepository.existsBySkuCodeAndQuantityInStockIsGreaterThanEqual(skuCode,quantityToCheck)
        }
        throw StockLevelException("The requested quantity of the item $skuCode is not available at this moment. Please try ordering a smaller quantity or only other items.")
    }

}
